framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -608,6 +613,19 @@ def writeToSolutionExport(self,traj):
     badValue = -1 #value to use if we don't have a value # TODO make this accessible to user?
     recent = self.counter['recentOptHist'][traj][0]
     for var in self.solutionExport.getVars():
+      # if this variable has indices, add them to the realization
+      indexes = self.solutionExport.getDimensions(var)[var]
+      if len(indexes):
+        # use the prefix to find the right realization
+        ## NOTE there will be a problem with unsynchronized histories!
```

Comment:
In the output space? most of the codes dump unsynchronized histories

Reply:
This is the solution export output, which I can't imagine having unaligned indices, but I'm sure we could come up with a situation where this would happen.

It might be a problem for codes delivering inconsistent histories, as well; that would be good followup work.  For the current needs, the model histories are required to be synchronized for other reasons.